documentation/C++style.md

```diff
@@ -19,6 +19,7 @@ is clear on usage, follow it.
   follow it.  [Google's](https://google.github.io/styleguide/cppguide.html)
   is pretty good and comes with lots of justifications for its rules.
 * Reasonable exceptions to these guidelines can be made.
+* Before submitting a pull request, review the [checkist below](#checklist).
```

Comment:
Do you want to matriculate from LLVM 7 at some point?

Reply:
I had a similar thought.  Maybe we should delay adding some (or all) of these items until we merge into LLVM.

Reply:
If this is about the version of clang-format, we should move to a newer one sometime soon. But it's good to specify the version so that everyone formats the same way.